### PR TITLE
txt.py and acq.py reorganization into io.py

### LINKED ISSUE #295: Reorganize I/O functions
## Summary

I find the `phys2bids.interfaces.[modality].populate_phys_input()` structure unnecessarily confusing. Why not have `phys2bids.io.load_matlab()`, `phys2bids.io.load_text()`, and `phys2bids.io.load_acq()`?

Note: this stems from https://github.com/physiopy/phys2bids/pull/292#issuecomment-676525617.

## Additional detail

I understand the potential benefits to having multiple functions with the same name (e.g., if you wanted to use `eval` and such to call functions), although, even in those cases, I don't know if it's worth the confusion. In `phys2bids`' case, however, the functions are called directly in the relevant code, so there isn't much benefit to this structure.

Also, `populate_phys_input()` as a function name mostly makes sense if you have multiple classes you might want to read in from files. In `phys2bids`' case, though, you really just want to read them into `BlueprintInput` objects (e.g., you're not going to read an acq file into a `BlueprintOutput` object), so I don't think there's any need to be so specific about the function name. A generic `load_` or `read_` should be sufficient.

## Next Steps

* Move the relevant functions to the same submodule (`phys2bids.io`). If you're concerned about their respective imports, you can just move the tool-specific imports into the function definitions.
* Rename the functions from `populate_phys_info()` to `load_matlab()`, `load_acq()`, and `load_text()` (or similar).
* Update tests and API documentation.

### PR DESCRIPTION
**Pull request by vinferrer**
Closes #295 

Merges `txt.py`and `acq.py` into `io.py`

## Proposed Changes

* Move the relevant functions to the same submodule (`phys2bids.io`). If you're concerned about their respective imports, you can just move the tool-specific imports into the function definitions.
* Rename the functions from `populate_phys_info()` to  `load_acq_ext()`, and `load_text_ext()` (or similar).
* Update tests and API documentation.

## Change Type

- [ ] `bugfix` (+0.0.1)
- [ ] `minor` (+0.1.0)
- [ ] `major`  (+1.0.0)
- [ ] `internal` (no version update)
- [ ] `documentation` (no version update)

## Checklist before review

- [x] I added everything I wanted to add to this PR.
- [x] \[Code or tests only\] I wrote/updated the necessary docstrings.
- [x] \[Code or tests only\] I ran and passed tests locally.
- [ ] \[Documentation only\] I built the docs locally.
- [x] My contribution is harmonious with the rest of the code: I'm not introducing repetitions.
- [x] My code respects the adopted style, especially linting conventions.
- [x] The title of this PR is explanatory on its own, enough to be understood as part of a changelog.
- [x] I added or indicated the right labels.

- [ ] I added information regarding the timeline of completion for this PR.

- [x] Please, comment on my PR while it's a draft and give me feedback on the development!

### CODE REVIEW

**Comment by eurunuela** [comment]
Thank you @vinferrer ! I don't have the bandwidth right now to review this but I will. 

I see you selected both `minormod` and `majormod`. How's that?

Also, we would be going under 90% code coverage, so we'll need some tests.

**Comment by vinferrer** [comment]
> Thank you @vinferrer ! I don't have the bandwidth right now to review this but I will.
> 
> I see you selected both `minormod` and `majormod`. How's that?
I don't know what to do about it.
> Also, we would be going under 90% code coverage, so we'll need some tests.
That's weird i didn't change any tests i just moved the functions around and modified the imports in the testing

**Comment by vinferrer** [comment]
I forgot to add the testing function, :laughing:

**Comment by eurunuela** [comment]
Now that I think of it, I believe this is an internal update. Not a `majormod` or a `minormod`. It really doesn't change the output of the package but makes our lives easier.

What do you think @tsalo @smoia ?

**Comment by smoia** [comment]
Hey y'all! Sorry, I noticed I didn't add the refactoring to the possible change types. But @vinferrer labelled it right: it's a refactoring.

"Internal" is for the infrastructure. Maybe we can change the label, now that I think about it...

**Comment by smoia** [comment]
BTW: I would get rid of the "interfaces" folder and rename `io.py` into `interfaces.py`. (It's not really an output function, is it?)

**Comment by tsalo** [comment]
An interface implies a class to me, with bidirectional interaction with the source. Having an input/output module also lets you put the `write_json`, `write_file`, etc. functions in the same place.

**Comment by eurunuela** [comment]
I like `io.py` better too.

**Comment by vinferrer** [comment]
so should i get rid of the folder then ?

**Comment by eurunuela** [comment]
> so should i get rid of the folder then ?

I would say yes. It does not make much sense to have the folder with just one file. And usually, `io.py` goes on the same level as `utils.py` for instance.

**Comment by vinferrer** [comment]
okay

**Comment by vinferrer** [comment]
I think this is ready for review then

**Review by smoia — COMMENTED** [review]
Generally LGTM, but unless everyone else is ok with it, could you remove `_ext` from the name of the functions?

**Comment by vinferrer** [comment]
> Generally LGTM, but unless everyone else is ok with it, could you remove `_ext` from the name of the functions?

Actually no, this is something that i put to remark that we are processing `extensions`  `.txt` is an extension, then we have formats like `acq` that can be in `.txt` or in `.acq` extensions

**Comment by vinferrer** [comment]
This is actually related to the next PR I talk about in #295. Because my goal is to have extension functions that will call in the end to the same process fucntion. This way we will reduce the amount of code

**Review by smoia — APPROVED** [review]
Alright, sounds good!
It's ready!

File: phys2bids/io.py [hunk 1]
```diff
@@ -1,18 +1,21 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
-"""phys2bids interface for txt files."""
-
+"""phys2bids interface for acqknowledge files."""
```
Comment (eurunuela) [inline phys2bids/io.py:4]: Why do you say it's the interface for acqknowledge files? We read txt files too.

**Review by eurunuela — APPROVED** [review]
LGTM! I just have a tiny comment.

File: phys2bids/io.py (hunk 1 above)
Comment (vinferrer) [reply phys2bids/io.py]: resolved

**Comment by smoia** [comment]
@vinferrer , you merged a PR without being the main reviewer (not a big deal), but also without passing the style check.
Please open a PR now to pass it before anyone else opens a new PR.

**Comment by smoia** [comment]
:rocket: PR was released in [`2.3.0`](https://github.com/physiopy/phys2bids/releases/tag/2.3.0) :rocket:

**Pull request merged**